### PR TITLE
feat: add Book entity and BookService

### PR DESCRIPTION
**Pull request by c25vdw**
not yet fleshed out. can we discuss and keep working on this?

### CODE REVIEW

File: app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    private BookService() {
+        bookCollection = FirebaseFirestore.getInstance().collection("books");
+    }
+
+    public static BookService getInstance() {
+        return instance;
+    }
+
+    public Task<QuerySnapshot> getBooksOfOwnerID(String uid) {
```
Comment (kdvuong) [inline app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java:23]: This can be combine with the one below to be getBooksByUID?

**Comment by kdvuong** [comment]
yep. I think having getBooks and addBook is enough for this PR. Everything else will be added whenever we need in future PR.

File: app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java (hunk 1 above)
Comment (c25vdw) [reply app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java:23]: how do we figure out the books that's borrowed by this user?
Option 1: have a borrower field in Book
Option 2: query for the user's requests whose book is borrowed.

Comment (kdvuong) [reply app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java:23]: I think option 2 is better

Comment (QuocTrungTran) [reply app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java:23]: I prefer option 2 as well

Comment (c25vdw) [reply app/src/main/java/com/cmput301f20t21/bookfriends/services/BookService.java:23]: We will need the Request entity for that functionality. In another PR, maybe?

**Review by kdvuong — APPROVED** [review]
lgtm

**Review by zhpeng811 — APPROVED** [review]
looks good for this PR

we need to add the optional description field in another PR

**Pull request merged**